internal/checks/checks.go

```
@@ -0,0 +1,171 @@
+package checks
+
+import (
+	"fmt"
+
+	"github.com/icinga/check-netgear/internal/utils"
+	"github.com/icinga/check-netgear/netgear"
+
+	"github.com/NETWAYS/go-check"
+	"github.com/NETWAYS/go-check/perfdata"
+	"github.com/NETWAYS/go-check/result"
+)
+
+// CPU
+func CheckCPU(cpuUsage float64, warn float64, crit float64) (result.PartialResult, int) {
+	status := utils.StatusByThreshold(cpuUsage, warn, crit)
+	partial := result.PartialResult{
+		Output: fmt.Sprintf("CPU Usage: %.2f%%", cpuUsage),
+	}
+	_ = partial.SetState(status)
```

Please make it a habit to not ignore errors like this. Instead, change the function return type like this because you don't have to return the `status` since you can access it via the `partial.GetStatus()` method if you need it.
```go
func CheckCPU(cpuUsage float64, warn float64, crit float64) (*result.PartialResult, error) {
	// ...
	if err := partial.SetState(status); err != nil {
		return nil, err
	}
	// ...
	return &partial, nil
}
```

The `//...` means the other stuff in that function that don't need to be changed.

Please adjust all the functions in this file just like this as well.